scvi/external/scar/_model.py

```diff
@@ -0,0 +1,310 @@
+import logging
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+import torch
+from anndata import AnnData
+from torch.distributions.multinomial import Multinomial
+from tqdm import tqdm
+
+from scvi import REGISTRY_KEYS
+from scvi._compat import Literal
+from scvi.data import AnnDataManager
+from scvi.data.fields import (
+    CategoricalJointObsField,
+    CategoricalObsField,
+    LayerField,
+    NumericalJointObsField,
+    NumericalObsField,
+)
+from scvi.model._utils import _init_library_size
+from scvi.model.base import (
+    ArchesMixin,
+    BaseModelClass,
+    RNASeqMixin,
+    UnsupervisedTrainingMixin,
+    VAEMixin,
+)
+from scvi.utils import setup_anndata_dsp
+
+from ._module import SCAR_VAE
+
+logger = logging.getLogger(__name__)
+
+
+class SCAR(
+    RNASeqMixin, ArchesMixin, VAEMixin, UnsupervisedTrainingMixin, BaseModelClass
+):
+    """
+    Ambient RNA removal in scRNA-seq data [Sheng22]_.
+    Original Github: https://github.com/Novartis/scar.
```

Comment:
If the model's are not parameter matched exactly I would comment that here.

Reply:
Agree, please comment it. We tested and optimised a unique set of parameters based on and for multiple data modalities.

Reply:
On the other hand, I'm curious how much more code is required to parameter match them exactly.

Reply:
By parameters, you guys mean both hyperparameters and model architecture parameters (latent dimension, hidden layers) right?

Reply:
yes, and then secondarily activations etc.

Reply:
Checklist of parameters matched:
- [x] nn_layer1=150
- [ ] nn_layer2=100 (set to 150 as well bc as far as I can tell scVI only allows uniform sizes for hidden layers)
- [x] latent_dim=15
- [x] dropout_prob=0
- [x] sparsity=0.9
- [ ] count_model="binomial" (I think we should go for the standard `zinb` in scVI for this one)
- [x] batch_norm (used in both encoder and decoder with `momentum=0.01, eps=0.001`
- [x] activation functions

Training params:
I'd propose sticking with the scVI training defaults here (using heuristics to determine number of epochs and kl warmup)
- [ ] kld_weight=1e-5
- [ ] epochs=800

Reply:
>  nn_layer2=100 (set to 150 as well bc as far as I can tell scVI only allows uniform sizes for hidden layers)

This is a fine compromise to me

>  count_model="binomial" (using nb - negative binomial from scVI)

This I think is an important distinction, can we make binomial default? Should just be using the torch binomial distribution?

> Training params:
I'd propose sticking with the scVI training defaults here (using heuristics to determine number of epochs and kl warmup)

This also seems fine to me

Reply:
In the latest commit I changed it to a binomial distribution (using torch's binomial) and it pretty much exactly recapitulates the original results (at least qualitatively -  see comparison below)

![download-91](https://user-images.githubusercontent.com/31646044/191548945-6c247e01-b8c4-4891-b805-aa1aadf9faf1.png)

Reply:
Great to see this @ricomnl. ZINB is more flexible than Binomial and Poisson distributions. It requires one more parameter to learn. We also found that Binomial distribution produces better results.

Reply:
Great to see this @ricomnl .

I approve these changes since you seem to recapitulate our results. I have not review all codes though.

Agreed. The default kld_weight in scVI seems to be 1.0. This should produce very good results for mRNA denoising with scAR, but not the optimal results for sgRNA assignment in scCRISPR-seq (which is not the case here). 
The epoch number with scVI's default setting should also produce good results for scAR. Generally, about 150 are enough in our test. 

> Training params:
I'd propose sticking with the scVI training defaults here (using heuristics to determine number of epochs and kl warmup)
  kld_weight=1e-5
 epochs=800